fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/Add.java

```diff
@@ -49,7 +49,7 @@ public Expression withChildren(List<Expression> children) {
 
     @Override
     public DecimalV3Type getDataTypeForDecimalV3(DecimalV3Type t1, DecimalV3Type t2) {
-        return (DecimalV3Type) DecimalV3Type.widerDecimalV3Type(t1, t2, false);
+        return (DecimalV3Type) DecimalV3Type.widerDecimalV3TypeForAddSub(t1, t2, false);
```

Comment:
do not change widerDecimalV3TypeForAddSub directly, use widerDecimalV3TypeForAddSub's result and plus 1 on its precision to create a new DecimalV3Type object